config/v1/types_infrastructure.go

```diff
@@ -505,9 +505,113 @@ type BareMetalPlatformStatus struct {
 	NodeDNSIP string `json:"nodeDNSIP,omitempty"`
 }
 
+type OpenStackAPIBGPPeer struct {
+	// asn is the ASN of the peer.
+	//
+	// +kubebuilder:validation:Required
+	ASN string `json:"asn"`
+
+	// ip is the IP address of the peer. It may be either IPv4 or IPv6.
+	//
+	// +kubebuilder:validation:Required
+	IP string `json:"ip"`
+}
+
+// OpenStackAPIBGPSpeaker species
+type OpenStackAPIBGPSpeaker struct {
+	// failureDomain is the name of a failure domain which this BGP
+	// configuration applies to.
+	//
```

Comment:
To use a failure domain the failure domain must be defined... <where>?

This context is useful to add to the comments IMO

Reply:
Added "A failure domain with that name must be defined in the OpenStack platform spec."